docarray/array/mixins/io/pushpull.py

```diff
@@ -1,24 +1,183 @@
+import json
 import os
+import os.path
 import warnings
+from collections import Counter
 from pathlib import Path
-from typing import Dict, Type, TYPE_CHECKING, Optional
+from typing import Dict, Type, TYPE_CHECKING, List, Optional, Any
+
+import hubble
+from hubble import Client
+from hubble.client.endpoints import EndpointsV2
+
 
 from docarray.helper import get_request_header, __cache_path__
 
 if TYPE_CHECKING:
     from docarray.typing import T
 
 
+def _get_length_from_summary(summary: List[Dict]) -> Optional[int]:
+    """Get the length from summary."""
+    for item in summary:
+        if 'Length' == item['name']:
+            return item['value']
+
+
 class PushPullMixin:
     """Transmitting :class:`DocumentArray` via Jina Cloud Service"""
 
     _max_bytes = 4 * 1024 * 1024 * 1024
 
+    @classmethod
+    @hubble.login_required
+    def cloud_list(cls, show_table: bool = False) -> List[str]:
```

Comment:
Am I understanding correctly that the intended usage is to do `DocumentArray.cloud_list()` (same for `cloud_delete()`)?
I am not sure if it is super intuitive to the user to interact with the `DocumentArray` class when what they are doing is communicating with a server. In my head this operation has not much to do with `DocumentArray`.

Could this be just a function that the user imports directly?

Reply:
I think static method looks more comprehensive

Reply:
I think it is better to be a `staticmethod` and not a `classmethod`. You do not use `cls` for anything right?